plugin.rb

```diff
@@ -521,4 +521,43 @@ def self.enqueue_assign_hooks(event, payload)
     end
   end
 
+  register_search_advanced_filter(/in:assigned/) do |posts|
+    if @guardian.can_assign?
+      posts.where("topics.id IN (
+        SELECT tc.topic_id
+        FROM topic_custom_fields tc
+        WHERE tc.name = 'assigned_to_id' AND
+                        tc.value IS NOT NULL
+        )")
+    end
+  end
+
+  register_search_advanced_filter(/in:not_assigned/) do |posts|
+    if @guardian.can_assign?
+      posts.where("topics.id NOT IN (
+        SELECT tc.topic_id
+        FROM topic_custom_fields tc
+        WHERE tc.name = 'assigned_to_id' AND
+                        tc.value IS NOT NULL
+        )")
+    end
+  end
+
+  register_search_advanced_filter(/assigned:(.+)$/) do |posts, match|
+    if @guardian.can_assign?
+      user_id = User.where(staged: false).where('username_lower = ? OR id = ?', match.downcase, match.to_i).pluck_first(:id)
```

Comment:
I liked `User.find_by_username` better so we don't duplicate the logic. I was thinking something like this

```ruby
register_search_advanced_filter(/assigned:(.+)$/) do |posts, match|
  if @guardian.can_assign?
    if user_id = User.find_by_username(match)&.id
      post.where("..... #{user_id}")
    end
  end
end
```

That way it will only filter when the username matches a real user and won't generate an error when it doesn't 😉